### PR TITLE
🌱[e2e] Install CNI using ClusterResourceSet and bump up CAPI to v0.3.9

### PR DESCRIPTION
**Pull request by sedefsavas**
**What this PR does / why we need it**:
This PR installs e2e CNIs using ClusterResourceSet.

This PR should hold until CAPI changes are merged: kubernetes-sigs/cluster-api#3332

Fixes: #1929 

/hold

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -137,16 +137,13 @@ test: ## Run tests
...
-
-$(OVERLAY_DIR)/kustomizeversions.yaml: $(OVERLAY_DIR) $(OVERLAY_SOURCE)/kustomizeversions.yaml
-	cp -f $(OVERLAY_SOURCE)/kustomizeversions.yaml $@
+$(OVERLAY_DIR): $(OVERLAY_DIR) $(OVERLAY_SOURCE)
+	cp -r $(OVERLAY_SOURCE) $@
 
 $(OVERLAY_DIR)/cluster-template.yaml: $(OVERLAY_DIR)
 	cp -f templates/cluster-template.yaml $@
 
-$(CONFORMANCE_CI_TEMPLATE): $(OVERLAY_DIR)/cluster-template.yaml $(ARTIFACTS)/templates $(KUSTOMIZE) $(OVERLAY_DIR)/kustomization.yaml $(OVERLAY_DIR)/kustomizeversions.yaml
```
[the remark this answers is not in the document]
Comment (randomvariable) [reply Makefile:149]: Revisiting, this looks like it was attempting to fix my brokenness. I'll fix

**Comment by randomvariable** [comment]
Rebasing after #1823 should resolve the makefile issues. Intention is to merge after 0.5.5 @sedefsavas

**Comment by sedefsavas** [comment]
@randomvariable When is the expected date for v0.5.5?

**Comment by randomvariable** [comment]
Very soon. Two PRs left at https://github.com/kubernetes-sigs/cluster-api-provider-aws/milestone/18

**Comment by ncdc** [comment]
@sedefsavas FYI `/hold` has to be on its own line without any additional text after it (this was previously held because of the `/hold` in the description)

**Comment by randomvariable** [comment]
@sedefsavas need to rebase this PR based on the e2e_new directory replacing the old "e2e" one.

**Comment by randomvariable** [comment]
If we can rebase this, and bump CAPI to 0.3.9 in this PR, that'll probably be everything we need.

**Comment by sedefsavas** [comment]
Changed capa-manager name as clusterctl requires canonical names in v0.3.9 release: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1028

**Comment by randomvariable** [comment]
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1816/files#diff-a1c3266d5eb66f7451bcaefc7d050fdfL280 failing, but unclear why.

**Comment by sedefsavas** [comment]
Yes, there is a problem with `multiple workload clusters` test, investigating.

**Comment by sedefsavas** [comment]
It was failing during deletion of a cluster, so maybe not a bug, this time it passes. Retesting.
/test pull-cluster-api-provider-aws-e2e

**Comment by randomvariable** [comment]
> It was failing during deletion of a cluster
in which case, #1887 should have resolved it.

/lgtm
/approve

**Pull request merged**